### ISSUE TITLE
@Defer not working

### ISSUE THREAD

**Issue by maxstue** [ChilliCream/hotchocolate#2946]
**Describe the bug**
When using @defer, like in [this post](https://chillicream.com/blog/2020/11/23/hot-chocolate-11) ,nothing happens and defer is not beeing used.

**To Reproduce**
Execute a query like this or a similar one like from the blog post, linked above.
```
{
  devices {
    id
    name
    ... @defer {
      status{
        ison
      }
    }
  }
}
```
![image](https://user-images.githubusercontent.com/20500253/105990291-8f4ba580-60a2-11eb-9707-a58f6ecb86d0.png)
There should be multiple packages comming from the server, but there aren't. 
The status field has even a Task delay so it takes longer than the rest of the query. Nothing changes if I make that delay higher.
The same request without @defer takes the same amount of time, so I asume @defer is not beeing used.

**Expected behavior**
Should receive two/multiple packages, just like in the picture from the blog post.

**Desktop (please complete the following information):**
 - OS: w10 latest
 - Version HC 11.0.9

**Additional context**
Am I using it wrong or do I need to add somthing to the graphql server?

**Comment by michaelstaib** [comment]
can you provide a repro project that shows that this does not work on your server?

**Comment by michaelstaib** [comment]
the query looks alright. the server does not need any setup so it should just work :)

**Comment by lTimeless** [comment]
[Startup.cs](https://github.com/SmartHub-Io/SmartHub/blob/feature/AddLightToggle/SmartHub/SmartHub.Api/Startup.cs)
[ServiceExtension with grapql configuration](https://github.com/SmartHub-Io/SmartHub/blob/feature/AddLightToggle/SmartHub/SmartHub.Api/Extensions/ServiceExtension.cs)
[the Field I testet @defer on](https://github.com/SmartHub-Io/SmartHub/blob/feature/AddLightToggle/SmartHub/SmartHub.Application/UseCases/Entity/Devices/ExtendObjectType/ExtendDeviceType.cs)

I hope these links help you  :)

**Comment by tobias-tengler** [comment]
I'm also facing a similar issue. I was trying out defer and added a "test" field to one of my ObjectTypes:
```csharp
descriptor.Field("test").Type<StringType>().Resolve(async context =>
{
    await Task.Delay(5000);
    return "Test";
});
```
Then I ran this query: 
```graphql
{
  entity(id: "600c7e36ad5476b22d5ba473") {
    id
    ... @defer {
      test
    }
  }
}
```
Banana Cake Pop shows that defer was indeed at work and submitted two patches.
![Banana Cake Pop](https://user-images.githubusercontent.com/45513122/109810930-7ea2c800-7c2a-11eb-8f94-9b80382d2e70.png)
However both of the patches come in at roughly the same time, with a 5 second difference to the initial query result.
I would've expected the query with the initial payload containing the id to arrive first and then after 5 seconds the "test" payload.
This does not seem like it's the expected behavior. Am I doing something wrong?

**Comment by michaelstaib** [comment]
We will have a look at this for the release end of next week....

**Comment by michaelstaib** [comment]
This is fixed with 12.0.0-rc.8

**Comment by tobias-tengler** [comment]
I'm seeing this issue again in any of the 13 previews. I can not reproduce it in a newly setup project though. What's the best way for me to debug this?
